wordcab_transcribe/services/asr_service.py

```diff
@@ -272,82 +272,93 @@ async def process_input(
+            if isinstance(task["diarization_result"], Exception):
+                self.gpu_handler.release_device(gpu_index)
+                return task["diarization_result"]
 
-        self.gpu_handler.release_device(gpu_index)  # Release the GPU
+            if diarization and task["diarization_result"] is None:
+                # Empty audio early return
+                return early_return(duration=duration)
 
-        asyncio.get_event_loop().run_in_executor(
-            None, functools.partial(self.process_post_processing, task)
-        )
+            if isinstance(task["transcription_result"], Exception):
+                self.gpu_handler.release_device(gpu_index)
+                return task["transcription_result"]
+            else:
+                if alignment and dual_channel is False:
+                    asyncio.get_event_loop().run_in_executor(
+                        None,
+                        functools.partial(
+                            self.process_alignment, task, gpu_index, self.debug_mode
+                        ),
+                    )
+                else:
+                    task["process_times"]["alignment"] = None
+                    task["alignment_done"].set()
 
-        await task["post_processing_done"].wait()
+            await task["alignment_done"].wait()
 
-        if isinstance(task["post_processing_result"], Exception):
-            return task["post_processing_result"]
+            if isinstance(task["alignment_result"], Exception):
+                logger.error(f"Alignment failed: {task['alignment_result']}")
+                # Failed alignment should not fail the whole request anymore, as not critical
+                # So we keep processing the request and return the transcription result
+                # return task["alignment_result"]
 
-        result: List[dict] = task.pop("post_processing_result")
-        process_times: Dict[str, float] = task.pop("process_times")
-        process_times["total"]: float = time.time() - start_process_time
+            self.gpu_handler.release_device(gpu_index)  # Release the GPU
+
+            asyncio.get_event_loop().run_in_executor(
+                None, functools.partial(self.process_post_processing, task)
+            )
 
-        del task  # Delete the task to free up memory
+            await task["post_processing_done"].wait()
 
-        return result, process_times, duration
+            if isinstance(task["post_processing_result"], Exception):
+                return task["post_processing_result"]
+
+            result: List[dict] = task.pop("post_processing_result")
+            process_times: Dict[str, float] = task.pop("process_times")
+            process_times["total"]: float = time.time() - start_process_time
+
+            del task  # Delete the task to free up memory
+
+            return result, process_times, duration
+        except Exception as e:
+            logger.error(f"Error processing input: {e}")
+            return e
```

Comment:
```suggestion
        except Exception as e:
            return e
```

The errors are already logged in the API endpoints:
<img width="494" alt="CleanShot 2023-08-31 at 08 24 28@2x" src="https://github.com/Wordcab/wordcab-transcribe/assets/50595514/7a641541-5f30-4afc-8797-7abc2338bf33">